### ISSUE TITLE
ERROR: column p.prokind does not exist: on Centos7, postgres9.6.10

### ISSUE THREAD

**Issue by petrheckoHims** [theory/pgtap#175]
Hello,

I'm trying to install pgTAB on Centos with Postgres9.6 running. When running `make installcheck`, the output is `33 of 33 tests failed`. In the output file `/home/ubuntu/pgtap/regression.diffs` I see bunch of errors like:
```
! ok 29 - another simple pass
--- 1,10 ----
  \unset ECHO
+ psql:sql/pgtap.sql:2497: ERROR:  column p.prokind does not exist
+ LINE 11:            p.prokind     AS kind,
+                     ^
+ HINT:  Perhaps you meant to reference the column "p.probin".
  1..29
! ERROR:  function findfuncs(name, unknown, unknown) does not exist
! LINE 2:     findfuncs('public'::name, '^test', 'this'),
!             ^
! HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

======================================================================

*** /var/lib/pgsql/pgtap/test/expected/performs_ok.out	2018-08-29 06:52:30.592489626 +0000
--- /var/lib/pgsql/pgtap/results/performs_ok.out	2018-08-29 07:27:34.646824278 +0000
***************
*** 1,26 ****
  \unset ECHO
  1..24
! ok 1 - simple select should pass
 ```

I'm seeing the same error when I'm trying to run our application which is using pgTAP and no tests run anymore. It still works locally on my Mac where I have installed postgres 9.6.9. The errors from our application:

```
[centos@ip-172-31-34-146 pth]$ rake clean_test
PTH_TARGET=all PTH_COVERAGE=none PTH_LINT=none
ERROR:  column p.prokind does not exist
HINT:  Perhaps you meant to reference the column "p.probin".
Error running: psql -qtAX --pset pager=off --set ON_ERROR_ROLLBACK=1 --set ON_ERROR_STOP=1 -h "localhost" -p "5599" -U "pth" -d "prod_db" -c "CREATE EXTENSION pgtap WITH SCHEMA pgtap"
[centos@ip-172-31-34-146 pth]$ which pg_config
/usr/pgsql-9.6/bin/pg_config
[centos@ip-172-31-34-146 pth]$ echo $PATH
/usr/pgsql-9.6/bin:/usr/local/rvm/gems/ruby-2.3.3/bin
```

Any idea about a workaround or what can be wrong with the centos installation?

**Comment by theory** [comment]
For some reason, your build thinks it's for Postgres 11; see 15858e2 for the use of the `prokind` column. Did you run `make && make install`? Did `compat/install-10.patch` get properly run when you ran `make`?

**Comment by theory** [comment]
Well that's not a permanent fix. More than likely, the version parsing code in the `Makefile` isn't quite right. Can you tell me the output of this command?

    echo 9.3.10 | grep -qE "([98]|10)[.]" && echo yes || echo no

**Comment by theory** [comment]
Better yet, try this:

     pg_config --version | awk '{print $2}' | grep -qE "([98]|10)[.]" && echo yes || echo no

It should output "yes". If not, try:

     pg_config --version | awk '{print $2}' | grep -qE "^([98]|10)[.]" && echo yes || echo no

**Comment by petrheckoHims** [comment]
Hi David,

I just tried to install pgtap on two separated Centos to verify my theory and I think I know what was my issue.

On both of the boxes, I installed postgres96 and verify both of the `pg_config` and looked fine:

```
[root@ip-172-31-35-54 centos]# pg_config --version
PostgreSQL 9.6.10
[root@ip-172-31-35-54 centos]# pg_config --version | awk '{print $2}' | grep -qE "([98]|10)[.]" && echo yes || echo no
yes

[root@ip-172-31-27-192 pgtap]# pg_config --version
PostgreSQL 9.6.10
[root@ip-172-31-27-192 pgtap]# pg_config --version | awk '{print $2}' | grep -qE "([98]|10)[.]" && echo yes || echo no
yes
```

The problem however was that I noticed from the history on the previous not-working centos box that after installing postgres96 with yum, I run `yum install postgresql-devel` instead of `yum install postgresql96-devel` - I was able to replicate the errors on my two new centos boxes. The one with the correct `postgresql96-devel` tools worked just fine, the one I installed `postgresql-devel` was throwing various errors. 

Thanks again for your help!
